src/product_logging/framework.rs

```diff
@@ -26,6 +31,65 @@ const SHUTDOWN_FILE: &str = "shutdown";
 /// File name of the Vector config file
 pub const VECTOR_CONFIG_FILE: &str = "vector.toml";
 
+/// Calculate the size limit for the log volume.
+///
+/// The size limit must be much larger than the sum of the given maximum log file sizes for the
+/// following reasons:
+/// - The log file rollover occurs when the log file exceeds the maximum log file size. Depending
+///   on the size of the last log entries, the file can be several kilobytes larger than defined.
+/// - The actual disk usage depends on the block size of the file system.
+/// - OpenShift sometimes reserves more than twice the amount of blocks than needed. For instance,
+///   a ZooKeeper log file with 4,127,151 bytes occupied 4,032 blocks. Then log entries were written
+///   and the actual file size increased to 4,132,477 bytes which occupied 8,128 blocks.
+///
+/// # Example
+///
+/// ```
+/// use stackable_operator::{
+///     builder::{
+///         PodBuilder,
+///         meta::ObjectMetaBuilder,
+///     },
+///     memory::{
+///         BinaryMultiple,
+///         MemoryQuantity,
+///     },
+/// };
+/// # use stackable_operator::product_logging;
+///
+/// const MAX_INIT_CONTAINER_LOG_FILES_SIZE: MemoryQuantity = MemoryQuantity {
+///     value: 1.0,
+///     unit: BinaryMultiple::Mebi,
+/// };
+/// const MAX_MAIN_CONTAINER_LOG_FILES_SIZE: MemoryQuantity = MemoryQuantity {
+///     value: 10.0,
+///     unit: BinaryMultiple::Mebi,
+/// };
+///
+/// PodBuilder::new()
+///     .metadata(ObjectMetaBuilder::default().build())
+///     .add_empty_dir_volume(
+///         "log",
+///         Some(product_logging::framework::calculate_log_volume_size_limit(
+///             &[
+///                 MAX_INIT_CONTAINER_LOG_FILES_SIZE,
+///                 MAX_MAIN_CONTAINER_LOG_FILES_SIZE,
+///             ],
+///         )),
+///     )
+///     .build()
+///     .unwrap();
+/// ```
+pub fn calculate_log_volume_size_limit(max_log_files_size: &[MemoryQuantity]) -> Quantity {
```

Comment:
I would have preferred a couple of tests to show what to expect from different file sizes.

Also what do you think about adding a cap to it. Say 1024Mi or something.

Reply:
+1 for a cap or some clever way such as (1/x with a min and max cap). Setting it to 3 times the amount seems pretty excessive.
I know we have pretty small log volumes currently, but how knows what happens in the future.
It would be great if this function is intended to work with volumes <= 1TB, as the limit should e.g. also be set for druid segment cache emptyDirs, and wasting 2/3 of 1 TB seems excessive

Reply:
Tests added in 4f71f38e9814e90ad24baa60d56912ebc252f22b

I do not know what reasonable values would be for the upper but as well for the lower limit.